sklearn/_loss/tests/test_loss.py

```diff
@@ -799,8 +799,8 @@ def test_loss_intercept_only(loss, sample_weight):
     if not loss.is_multiclass:
         y_true = loss.link.inverse(np.linspace(-4, 4, num=n_samples))
     else:
-        y_true = np.arange(n_samples).astype(float) % loss.n_classes
-        y_true[::5] = 0  # exceedance of class 0
+        y_true = np.arange(n_samples).astype(np.float64) % loss.n_classes
+        y_true[::5] = 0  # exceedance of null events in the target
```

Comment:
This is not really related to the PR but I fixed those details when spotting them.

In general, I think we should not use Python types as NumPy dtype specs because the precision level Python types is not explicit and could potentially lead to platform specific variations.